src/clj/puppetlabs/services/jruby_pool_manager/impl/reference_pool.clj

```diff
@@ -0,0 +1,86 @@
+(ns puppetlabs.services.jruby-pool-manager.impl.reference-pool
+  (:require [puppetlabs.services.protocols.jruby-pool :as pool-protocol]
+            [puppetlabs.services.jruby-pool-manager.impl.jruby-agents :as jruby-agents]
+            [puppetlabs.services.jruby-pool-manager.impl.jruby-internal :as jruby-internal]
+            [puppetlabs.services.jruby-pool-manager.jruby-schemas :as jruby-schemas]
+            [clojure.tools.logging :as log]
+            [puppetlabs.i18n.core :as i18n])
+  (:import (puppetlabs.services.jruby_pool_manager.jruby_schemas ReferencePool)))
+
+(extend-type ReferencePool
+  pool-protocol/JRubyPool
+
+  (fill
+    [this]
+    (jruby-agents/add-instance this 1))
```

Comment:
This is synchronous, which is a departure from the old implementation, which spun off new threads to create instances. Since only one instance is being added, and no other pool operations can proceed while there is no instance, it seemed okay to do this, but I only did it to make things easier to test, which seems like a bad reason. Thoughts on this?

Reply:
probably ok?

Reply:
Is it possible to insert some kind of flow control as an initialize-instance-fn in the test vs making this synchronous?

fwiw, I'm unconcerned about the pool operations being synchronous so much as the calling puppetserver start up code becoming synchronous.